.golangci.yml

```diff
@@ -0,0 +1,205 @@
+run:
+  timeout: 10m
+  issues-exit-code: 1
+  tests: true
+
+linters:
+  enable:
+    - bidichk                 # find dangerous bidi unicode symbols
+    - bodyclose               # find unclosed http response bodies
+    - dogsled                 # checks for too many ignored arguments
+    - durationcheck           # verifies whether durations are multiplied, usually a mistake
+    - errcheck                # find unchecked errors
+    - errorlint               # finds misuses of errors
+    - exportloopref           # check for exported loop vars
+    - gocritic                # checks for style, performance issues, and common programming errors
+    - godot                   # dots for everything
+    - err113                  # check error expressions
+    - gofmt                   # sanity check formatting
+    - goprintffuncname        # checks that printf-like functions are named with `f` at the end [fast: true, auto-fix: false]
+    - govet                   # check standard vet rules
+    - importas                # verify that imports are consistent
+    - ineffassign             # find ineffective assignments
+    - makezero                # verifies that slices use defaults
+    - misspell                # check spelling
+    - nakedret                # check for naked returns
+    - nilerr                  # checks for misuses of `if err != nil { return nil }`
+    - noctx                   # finds locations that should use context
+    - revive                  # check standard linting rules
+    - tenv                    # ensure we use t.SetEnv instead of os.SetEnv
+    - unconvert               # remove unnecessary conversions
+    - wastedassign
+    #TODO#- nilnil            # needs fixes; checks that we don't return a nil and no-error
+    #TODO#- forcetypeassert   # needs work to replace unchecked interface type assertion
+    #TODO#- gochecknoglobals  # needs work to remove globals
+    #TODO#- gochecknoinits    # needs work to remove init()
+    #TODO#- gofumpt           # not sure whether it's useful
+    #TODO#- nestif            # looks useful, however needs work
+    #TODO#- prealloc          # easy optimizations
+    #TODO#- unparam           # check for unused parameters
+    #TODO#- whitespace        # checks for leading/trailing newlines
+  disable:
+    - nolintlint              # checks that nolint directives are correct
+    - staticcheck      # we already use staticcheck separately
```

Comment:
Oh, forgot, please add this into the enable section.

We run staticcheck separately in other repos, because the golangci-lint sometimes lags behind with the version.

Also, staticcheck is the most useful part of the checks.